lib/portage/update.py

```diff
@@ -175,18 +175,15 @@ def grab_updates(updpath, prev_mtimes=None):
 		raise
 	if prev_mtimes is None:
 		prev_mtimes = {}
-	# validate the file name (filter out CVS directory, etc...)
-	mylist = [myfile for myfile in mylist if len(myfile) == 7 and myfile[1:3] == "Q-"]
-	if len(mylist) == 0:
-		return []
-
-	# sort by (year, quarter)
-	mylist.sort(key=lambda x: (x[3:], x[:2]))
 
 	update_data = []
 	for myfile in mylist:
+		if myfile.startswith("."):
+			continue
 		file_path = os.path.join(updpath, myfile)
 		mystat = os.stat(file_path)
+		if not stat.S_ISREG(mystat.st_mode):
+			continue
 		if update_data or \
```

Comment:
I think without sorting, the logic of this following `if` statement makes even less sense than it did before. If `prev_mtimes` is given, then only the files with different mtimes should be considered.

Adding any other "later" files to the list doesn't make sense, because PMS doesn't specify any order for processing of these files, so there is no concept of "later", in the first place.

Reply:
Makes sense. I'm recombining the PRs now, I'll fix it after that.